src/reachy_mini/media/audio_gstreamer.py

```diff
@@ -60,37 +60,59 @@ def __init__(self, log_level: str = "INFO") -> None:
 
     def _init_pipeline_record(self, pipeline: Gst.Pipeline) -> None:
         self._appsink_audio = Gst.ElementFactory.make("appsink")
-        caps = Gst.Caps.from_string(
-            f"audio/x-raw,rate={self.SAMPLE_RATE},channels={self.CHANNELS},format=F32LE,layout=interleaved"
+        
+        # 1. Define the target caps (What we WANT: 16k, F32LE)
+        caps_string = (
+            f"audio/x-raw,rate={self.SAMPLE_RATE},"
+            f"channels={self.CHANNELS},"
+            f"format=F32LE,layout=interleaved"
         )
+        caps = Gst.Caps.from_string(caps_string)
+        
+        # Configure AppSink
         self._appsink_audio.set_property("caps", caps)
-        self._appsink_audio.set_property("drop", True)  # avoid overflow
+        self._appsink_audio.set_property("drop", True)
         self._appsink_audio.set_property("max-buffers", 200)
 
-        audiosrc: Optional[Gst.Element] = None
+        # 2. Select Source (Hardware vs Auto)
         if self._id_audio_card == -1:
-            audiosrc = Gst.ElementFactory.make("autoaudiosrc")  # use default mic
+            self.logger.info("Using autoaudiosrc (Default System Device)")
+            audiosrc = Gst.ElementFactory.make("autoaudiosrc")
         else:
+            self.logger.info(f"Using alsasrc device plughw:{self._id_audio_card},0")
             audiosrc = Gst.ElementFactory.make("alsasrc")
-            audiosrc.set_property("device", f"hw:{self._id_audio_card},0")
+            audiosrc.set_property("device", f"plughw:{self._id_audio_card},0")
 
+        # 3. Create Processing Elements
         queue = Gst.ElementFactory.make("queue")
         audioconvert = Gst.ElementFactory.make("audioconvert")
         audioresample = Gst.ElementFactory.make("audioresample")
-
-        if not all([audiosrc, queue, audioconvert, audioresample, self._appsink_audio]):
-            raise RuntimeError("Failed to create GStreamer elements")
-
-        pipeline.add(audiosrc)
-        pipeline.add(queue)
-        pipeline.add(audioconvert)
-        pipeline.add(audioresample)
-        pipeline.add(self._appsink_audio)
-
-        audiosrc.link(queue)
-        queue.link(audioconvert)
-        audioconvert.link(audioresample)
-        audioresample.link(self._appsink_audio)
+        
+        # 4. CRITICAL FIX: Explicit CapsFilter
+        # This forces the audioresample to actually work before data hits the sink
+        capsfilter = Gst.ElementFactory.make("capsfilter")
```

Comment:
There is no need for the capsfilter here since they are already set to the appsink